### DISCUSSION TITLE
Need a help in to retrieve DCIR power losses in HFSS 3d Layout using pyaedt

### DISCUSSION THREAD

**Discussion by nnetake** [ansys/pyaedt#3964] [Q&A]
I am working on a pyaedt-based workflow of hfss 3d layout and Icepak co-simulation. I am from Icepak side.
I have one solved DCIR setup in hfss 3d layout and want to retrieve all layers power losses using pyaedt/native script to be used down the line in Icepak. Do you have any idea how to do it?
![image](https://github.com/ansys/pyaedt/assets/118507703/3e15400c-240e-46b8-818c-e6782495c225)

**Comment by Samuelopez-ansys** [comment]
I was checking, and it seems this power is calculated internally and is not exposed in the API.  I think you can just use the Fields Calculator to get the Power density on each net of the layer and the sum of all of them. What do you think? With a simple loop, you should be able to do it. So as I said, getting this number directly is not possible, and you could submit an enhancement request (internally in Ansys) to have it in future AEDT versions.

**Reply by Samuelopez-ansys** [reply]
If you want it as a new feature in PyAEDT, please submit an Issue, to have in in the backlog of new enhancements. Thanks

**Reply by nnetake** [reply]
@Samuelopez-ansys I tried with a field calculator, in geometry, it does not have any volumes but only surfaces. If I integrate power density over surfaces it does not give the correct value.
![image](https://github.com/ansys/pyaedt/assets/118507703/95c03665-33c0-4c6a-8f96-d8f2bd7b6161)

**Reply by Samuelopez-ansys** [reply]
@nnetake Did you try to sum up all surfaces (nets) of one layer? I think HFSS 3DLayout is not considering any volume, it is taking the layer width and the surface to compute the total power very likely, but t his value is computed internally. Maybe @ring630 or @svandenb-dev have any idea how to get this data.

**Reply by nnetake** [reply]
@Samuelopez-ansys Yes I tried to get it over all nets and vias, it shows incorrect value:
![image](https://github.com/ansys/pyaedt/assets/118507703/f01819d8-c3a0-4df7-8550-209f0cab4807)
![image](https://github.com/ansys/pyaedt/assets/118507703/2f508e80-f5aa-4c17-96aa-4bca8c113f33)

**Comment by hui-zhou-a** [comment]
@Samuelopez-ansys @nnetake the total power loss can be found in .totalpwr file in result folder.
![image](https://github.com/ansys/pyaedt/assets/27995305/0a846566-cb4b-4322-9ebd-43fd900ab2fc)
